### PR TITLE
Adding a test case for TimeZoneInfo.GetSystemTimeZones.

### PR DESCRIPTION
**Pull request by eerhardt**
@ellismg, @tarekgh

### CODE REVIEW

File: src/System.Runtime/tests/System/TimeZoneInfo.cs [hunk 1]
```diff
@@ -1745,6 +1746,44 @@ public static void TestLisbonDaylightSavingsWithNoOffsetChange(string dateTimeSt
...
+        Assert.Contains(timeZones, t => t.Id == s_strAmsterdam);
+        Assert.Contains(timeZones, t => t.Id == s_strRussian);
+        Assert.Contains(timeZones, t => t.Id == s_strLibya);
+        Assert.Contains(timeZones, t => t.Id == s_strCatamarca);
+        Assert.Contains(timeZones, t => t.Id == s_strLisbon);
+        Assert.Contains(timeZones, t => t.Id == s_strNewfoundland);
+
+        // ensure the TimeZoneInfos are sorted by BaseUtcOffset and then DisplayName.
+        TimeZoneInfo previous = timeZones[0];
+        for (int i = 1; i< timeZones.Count; i++)
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/TimeZoneInfo.cs]: Nit: space before <

**Comment by stephentoub** [comment]
LGTM

File: src/System.Runtime/tests/System/TimeZoneInfo.cs [hunk 2]
```diff
@@ -1745,6 +1746,44 @@ public static void TestLisbonDaylightSavingsWithNoOffsetChange(string dateTimeSt
...
+        {
+            TimeZoneInfo current = timeZones[i];
+            int baseOffsetsCompared = current.BaseUtcOffset.CompareTo(previous.BaseUtcOffset);
```
Comment (tarekgh) [inline src/System.Runtime/tests/System/TimeZoneInfo.cs]: I think the documentation says the returned list is sorted by the display name and not the Base utc offset. 

https://msdn.microsoft.com/en-us/library/system.timezoneinfo.getsystemtimezones(v=vs.110).aspx

Comment (tarekgh) [inline src/System.Runtime/tests/System/TimeZoneInfo.cs]: I mean it is not guaranteed to have baseOffsetsCompared >= 0. right?

Comment (eerhardt) [inline src/System.Runtime/tests/System/TimeZoneInfo.cs]: I think the documentation is incorrect.  The code also has a comment that the TimeZones are sorted by DisplayName, but the TimeZoneInfoComparer sorts them by BaseUtcOffset, and then by DisplayName.
See https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/TimeZoneInfo.cs#L5479.

Comment (tarekgh) [inline src/System.Runtime/tests/System/TimeZoneInfo.cs]: Yes I checked that and you are right. I'll contact the doc team to fix it

**Comment by tarekgh** [comment]
LGTM

File: src/System.Runtime/tests/System/TimeZoneInfo.cs [hunk 3]
```diff
@@ -1715,11 +1716,11 @@ public static void TestLisbonDaylightSavingsWithNoOffsetChange(string dateTimeSt
     // using non-UTC date times in this test to get some converage for non-UTC date times
```
Comment (joshfree) [inline src/System.Runtime/tests/System/TimeZoneInfo.cs]: nit: typo

**Comment by joshfree** [comment]
:+1:

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Pull request merged**